gtsfm/evaluation/metrics.py

```diff
@@ -90,8 +90,11 @@ def __init__(
         self._name = name
         if data is not None:
             # Cast to a numpy array
-            if isinstance(data, list) and all(isinstance(x, int) for x in data):
-                data = np.array(data, dtype=np.int_)
+            if isinstance(data, list):
```

Comment:
just curious, where was this issue showing up?

Reply:
Conversion should already be automatic from None -> NaN if we are doing integer conversions for float:

```
>>> import numpy as np
>>> np.array([None, 1, 2, None])
array([None, 1, 2, None], dtype=object)
>>> x = np.array([None, 1, 2, None])
>>> y = np.array(x, dtype=np.float32)
>>> y
array([nan,  1.,  2., nan], dtype=float32)
```

but I guess we are using all ints here, and not casting to float32 with these int args?

Reply:
I dont think we want to cast all metrics to float. there are some metrics that are assumed to be `int` for good reason. for example: the track length histogram in the report relies on all numbers being ints.